src/main/java/org/apache/maven/buildcache/hash/Zah.java

```diff
@@ -20,43 +20,55 @@
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
 import net.openhft.hashing.LongHashFunction;
 
+import static java.nio.channels.FileChannel.MapMode.READ_ONLY;
+import static java.nio.file.StandardOpenOption.READ;
+
 /**
- * XX
+ * Zero-Allocation-Hash based factory
  */
-public class XX implements Hash.Factory {
+public class Zah implements Hash.Factory {
+
+    private final String name;
+    private final LongHashFunction hash;
+    private final boolean useMemoryMappedBuffers;
 
-    static final LongHashFunction INSTANCE = LongHashFunction.xx();
+    public Zah(String name, LongHashFunction hash, boolean useMemoryMappedBuffers) {
```

Comment:
-1 on the boolean parameter. What if you need to distinguish further? Better use a String or Enum.